cmd/gossamer/config.go

```diff
@@ -240,10 +251,57 @@ func setDotRPCConfig(ctx *cli.Context, cfg *dot.RPCConfig) {
 	}
 
 	log.Debug(
-		"[cmd] RPC configuration",
+		"[cmd] rpc configuration",
 		"enabled", cfg.Enabled,
 		"port", cfg.Port,
 		"host", cfg.Host,
 		"modules", cfg.Modules,
 	)
 }
+
+// updateFromGenesis updates the configuration based on the genesis file values
+func updateFromGenesis(cfg *dot.Config) {
+
+	// load Genesis from genesis configuration file
+	gen, err := genesis.LoadGenesisFromJSON(cfg.Global.Genesis)
+	if err != nil {
+		log.Error("[cmd] failed to load genesis from file", "error", err)
+		return // exit
+	}
```

Comment:
unless you are doing `init`, the genesis file isn't relevant anymore, not sure why it's in the config. if you are running gossamer when it's already initialized, you should be retrieving the genesis data using `state.LoadGenesisData`

Reply:
This is just loading the genesis properties so that they can be added to the configuration. Let me know if this is still confusing or if it still needs to be fixed.

Reply:
if this is being called from `initAction` this is ok. if this is being called from `gossamerAction` it should be reading the genesis data from the database

Reply:
ok, with the added check if the genesis flag is being passed to `gossamerAction`, sounds like this is ok as is then. :+1:

Reply:
not really, since this still allows incorrect genesis data to be read (ie. not what the node was initialized with)

except for the `init` case, the genesis data should *only* be read from the database

Reply:
the issue explicity says that the genesis data should be loaded with `LoadGenesisData`